contracts/aave/AaveV2Contract.sol

```diff
@@ -0,0 +1,173 @@
+    }
+
+    //----------------------------//
+    //     Mutation Functions     //
+    //----------------------------//
+
+    /**
+     * @dev Only owner can add new assets that the contract will support for staking
+     * @param _asset will be address of the asset
+     * @param _priceOracle will the chainlink asset price oracle
+     * @return event AddAsset(_asset: 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2, _priceOracle: 0xAc559F25B1619171CbC396a50854A3240b6A4e99)
+     */
+    function addDepositAsset(
+        address _asset,
+        address _priceOracle
+    ) public onlyOwner isValid( _asset) {
+        depositAssets[_asset] = _priceOracle;
+        emit AddAsset(_asset,_priceOracle);
+    }
+
+    /**
+     * @dev Users will deposit their assets in pool
+     * @param _asset will be asset name like ETH or BTC
+     * @param _amount will be the deposited amount
+     */
+    function deposit( address _asset,uint256 _amount) public nonReentrant payable {
+        require(
+            depositAssets[_asset] != address(0),
+            Errors.ASSET_NOT_ALLOWED
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+
+        if (msg.value > 0) {
+            WETHGATEWAY.depositETH{value: msg.value}(
+                address(POOL),
+                address(this),
+                0
+            );
+        } else {
+            require(_amount > 0, Errors.ZERO_AMOUNT);
+            IERC20 supplyAsset = IERC20(_asset);
+            supplyAsset.transferFrom(msg.sender, address(this), _amount);
+            supplyAsset.approve(address(POOL), _amount);
+            POOL.deposit(address(supplyAsset), _amount, address(this), 0);
+        }
+    }
+
+    /**
+     * @dev Cruize will borrow on behalf of user
+     * @param _asset will be asset name like ETH or BTC
+     */
+    function borrow(address _asset) public nonReentrant {
+        (, , uint256 availableBorrowsETH, , , ) = POOL.getUserAccountData(
+            address(this)
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+        uint256 availableBorrowAmountIn18Decimals = (nomalize(
+            uint256(assetPrice)
+        ) * availableBorrowsETH) / 1e18;
+        uint256 availableBorrowAmountIn6Decimals = availableBorrowAmountIn18Decimals /
+                1e12;
+        uint256 borrowAmount = (availableBorrowAmountIn6Decimals *
+            borrowRatio) / 1000;
+        POOL.borrow(address(BORROW_ASSET), borrowAmount, 2, 0, address(this));
+    }
+
+    /**
+     * @dev Cruize pool will repay the debt amount
+     */
+    function repay() public nonReentrant {
+        BORROW_ASSET.approve(address(POOL), type(uint256).max);
+        POOL.repay(address(BORROW_ASSET), type(uint256).max, 2, address(this));
+    }
+
+    /**
+     * @dev Criuze will withdraw collateral amount
+     * @param _asset will be asset name like ETH or BTC
+     */
+    function withdraw(address _asset) public nonReentrant {
```

Comment:
and what does this withdrawal will do? will it withdraw in the user's account or in Sc address according  to the code it will withdraw in contract why did you make it public it means any one can call it